framework/include/utils/ParallelUniqueId.h

```diff
@@ -32,20 +37,30 @@ class ParallelUniqueId
   ParallelUniqueId()
   {
 #ifdef LIBMESH_HAVE_TBB_API
-    ids.pop(id);
-#else
-#ifdef LIBMESH_HAVE_PTHREAD
-    id = Threads::pthread_unique_id();
+    _ids.pop(id);
+#elif LIBMESH_HAVE_OPENMP
+    id = omp_get_thread_num();
+#elif LIBMESH_HAVE_PTHREAD
+    Threads::spin_mutex::scoped_lock lock(_pthread_id_mutex);
+
+    if (_ids.empty())
+      throw MooseException("No Thread IDs available in ParallelUniqueID. Did you forget to initialize()?");
+
+    id = _ids.front();
+    _ids.pop();
```

Comment:
This means a ParallelUniqueId object can be created only _one_ per thread if pthreads is being used. Is that how we use that class?